protocol/badgegenerator/server.go

```diff
@@ -122,10 +134,11 @@ func (s *Server) GenerateBadge(ctx context.Context, req *pairingtypes.GenerateBa
 		return nil, err
 	}
 	badge := pairingtypes.Badge{
-		CuAllocation: uint64(projectData.EpochsMaxCu),
+		CuAllocation: uint64(projectData.EpochsMaxCu) * (s.GetVirtualEpoch() + 1), // add additional CU due to emergency mode
```

Comment:
if the handling of badge cu calculation is on the SDK and the badge server is only monitoring the number of virtual epochs that passed, we do not need to also apply the cu here as the SDK can apply the cu limits on its own